packages/numeric-input/src/NumericInput.js

```diff
@@ -0,0 +1,178 @@
+import SpinnerPresenter from "./presenters/SpinnerPresenter.js";
+
+const variantTypes = ["line", "box"];
+
+export default class NumericInput extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    onBlur: PropTypes.func,
+    /**
+     * Called after user changes the value of the field
+     */
+    onChange: PropTypes.func,
+    /**
+     * Called when user puts focus onto the field
+     */
+    onFocus: PropTypes.func,
+    /**
+     * Called as user changes the value of the field
+     */
+    onInput: PropTypes.func,
+    /**
+     * How much you want the arrows to move up or down
+     */
+    placeholder: PropTypes.string,
+    /**
+     * Initial text to display in the input
+     */
+    step: PropTypes.number,
+    /**
+     * Adds custom/overriding styles
+     */
+    stylesheet: PropTypes.func,
+    /**
+     * Starting value for the input
+     */
+    value: PropTypes.number,
+    /**
+     * The visual variant of the numeric input
+     */
+    variant: PropTypes.oneOf(variantTypes),
+    onClick: PropTypes.func,
+    disabled: PropTypes.bool,
+    defaultValue: PropTypes.number
+  };
+
+  static defaultProps = {
+    step: 1,
+    variant: "line"
+  };
+
+  render() {
+    const {
+      disabled,
+      onBlur,
+      onChange,
+      onFocus,
+      onInput,
+      placeholder,
+      step,
+      stylesheet,
+      value,
+      defaultValue: initialValue,
+      variant,
+      onClick,
+      ...otherProps
+    } = this.props;
+
+    const numericInputStylesheet = (styles, props, themeData) => {
+      const numericInputStyles = customStylesheet(styles, props, themeData);
+      return stylesheet
+        ? stylesheet(numericInputStyles, props, themeData)
+        : numericInputStyles;
+    };
+
+    return (
+      <div>
+        <ControlBehavior
+          onBlur={onBlur}
+          onFocus={onFocus}
+          onMouseDown={this.props.onMouseDown}
+          onMouseEnter={this.props.onMouseEnter}
+          onMouseLeave={this.props.onMouseLeave}
+          onMouseUp={this.props.onMouseUp}
+        >
+          {({
+            hasFocus,
+            hasHover,
+            isPressed,
+            onBlur: handleBlur,
+            onFocus: handleFocus,
+            onMouseDown: handleMouseDown,
+            onMouseEnter: handleMouseEnter,
+            onMouseLeave: handleMouseLeave,
+            onMouseUp: handleMouseUp
+          }) => (
+            <SpinnerBehavior
+              onClick={onClick}
+              value={value}
+              onChange={onChange}
+              initialValue={initialValue}
+              step={step}
+              disabled={disabled}
+            >
+              {({
+                handleClick,
+                handleChange,
+                increment,
+                decrement,
+                value: controlledValue,
+                setValue
+              }) => {
+                const onDirectChange = event => {
```

Comment:
This can't be moved into the `SpinnerBehavior`? If not would it make sense to move this up in scope with other `const`, just looks odd to all of a sudden have an event handler here.

Reply:
I tried moving it into SpinnerBehavior but then the number wouldn't be updated anymore. And I'm not sure if I can move it out of the current scope since it needs the `setValue` function from `SpinnerBehavior`

Reply:
I made a previous comment in `SpinnerBehavior` that `handleChange` was not defined. I think you can move the `onDirectChange` function to the `SpinnerBehavior` and use that as the `handleChange` function, then pass it back down to the `Input` instead of using `onDirectChange` from this file.